src/dataflow/concatenate_h5mu/script.py

```diff
@@ -279,6 +281,13 @@ def concatenate_modality(
     if uns_merge_mode == "make_unique":
         concatenated_data = make_uns_keys_unique(mod_data, concatenated_data)
 
+    # Remove obsp keys that are not in par["obsp_keys"]
+    if par["obsp_keys"]:
+        # Keep only the obsp keys that are specified in par["obsp_keys"]
+        keys_to_remove = set(concatenated_data.obsp.keys()) - set(par["obsp_keys"])
+        for key in keys_to_remove:
+            del concatenated_data.obsp[key]
+
```

Comment:
I would remove the `obsp` keys from each individual anndata _before_ running `anndata.concat`. That way the overhead of concatenating _all_ of the obsp slots is somewhat reduced.